conans/client/generators/virtualenv.py

```diff
@@ -45,27 +45,31 @@ def format_values(self, flavor, variables):
 
         ret = []
         for name, value in variables:
+            # deactivate values
+            old_value = os.environ.get(name, "")
+            deactivate_value = "\"%s\"" % old_value if quote_full_value or quote_elements else old_value
+
             # activate values
+            placeholder = self._variable_placeholder(flavor, name)
             if isinstance(value, list):
-                placeholder = self._variable_placeholder(flavor, name)
                 if enable_space_path_sep and name in self.append_with_spaces:
                     # Variables joined with spaces look like: CPPFLAGS="one two three"
-                    value = " ".join(value+[placeholder])
+                    value = " ".join(value + [placeholder])
                     value = "\"%s\"" % value if value else ""
                 else:
                     # Quoted variables joined with pathset may look like:
                     # PATH="one path":"two paths"
                     # Unquoted variables joined with pathset may look like: PATH=one path;two paths
                     value = ["\"%s\"" % v for v in value] if quote_elements else value
-                    value = path_sep.join(value+[placeholder])
-            else:
-                # single value
+                    value = path_sep.join(value + [placeholder])
+            else:  # single value
+                if old_value != value and old_value is not "" and old_value in value:
+                    # Add placeholder instead of old value
+                    new_value = value.replace(old_value, "")
```

Comment:
The old value should be replaced with the placeholder. In the same place. Otherwise, the variable could mess with the order or even break the value consistency.